dev-requirements.txt

```diff
@@ -6,8 +6,8 @@ tornado==5.1.1
 PySocks==1.6.8
 pkginfo==1.4.2
 pytest-timeout==1.3.1
-pytest==3.8.2
-pluggy==0.8.0
+pytest==4.0.0
```

Comment:
Can we unpin this to the most recent version with this current PR?

Reply:
Not yet there is still some work pending
1. Deleting the unittest.TestCase base class everywhere
2. Replacing all def setUp with def setup_method
3. Delete all other references to unittest (imports, if __name__ == "__main__": unittest.main())
4. declare *TestCase classes as test classes
5. call super().setup_class() manually

Reply:
However we can remove the pluggy dependency as urllib3 does not use directly and the latest version now works. @sethmlarson Maybe that's what you were referring to?

Reply:
I'm just wondering what work still needs to be done for us to be using the latest pytest. :) This is a good start down that road though. Is this PR ready for a larger review or?

Reply:
There are few formatting issues also

Reply:
@sethmlarson Thanks for asking! Yeah, I don't think you need to look at this until my comments are addressed. There are not many of them left!